fetcher.go

```diff
@@ -346,10 +346,8 @@ func (p *pool) fetchResource(ctx context.Context, from string, resource string,
 	wrapped := TrackingReader{resp.Body, time.Time{}, 0}
 	err = cb(resource, &wrapped)
 	received = wrapped.len
-	// drain body so it can be re-used.
-	_, _ = io.Copy(io.Discard, resp.Body)
-
 	if err != nil {
+		resp.Body.Close()
```

Comment:
this will double-close. i think we just don't have this line

Reply:
Fixed